### ISSUE TITLE
🚀 Feature: Decouple Nginx from official Docker

### ISSUE THREAD

**Issue by avesst** [stonith404/pingvin-share#318]
### 🔖 Feature description

I'm proposing the decoupling of Nginx from the official Docker image and instead relying on a separate image which can be used in conjunction with the main image with Docker Compose.

I can't really see the purpose of including Nginx in the official image for the application - other than making the install process quicker, which can be achieved with Docker Compose. I can however see a couple of limitations - such as the one mentioned in the README and that it adds unecessary complexity. It also kinda goes against the Docker philosophy of "one service per container".

### 🎤 Pitch

It will make it easier to work with.

**Comment by maetthew** [comment]
I would like to add that this application looks extremely promising and seems to tick almost all the boxes of something that I've been looking for a while.

I don't know enough JavaScript/TypeScript to contribute in that area but if this is deemed a good idea I would be happy to put in the work to make it happen.

**Comment by stonith404** [comment]
@matthew Yeah, I understand your point, and I create separate Docker images for my other projects as well. In the beginning of Pingvin Share, I also had two different images, but the self-hosted community suggested that I put everything in one container to simplify the setup.

**For everyone, feel free to comment on this issue with your thoughts.**

Nevertheless, it would be possible to create both an all-in-one Docker image and also two separate ones.

**Comment by jackadam1981** [comment]
建议放一个docker-compose.yaml，会比一个docker image方便吧？

**Comment by stonith404** [comment]
@jackadam1981 Please compose your comments in English as the most of use don't understand Chinese.

**Comment by aarondoet** [comment]
How about decoupling it but providing a minimal production ready docker-compose.yml that has a reverse proxy configured?

**Comment by stonith404** [comment]
@aarondoet If we decouple the reverse proxy from the container we should also create separate images for the backend and frontend in my opinion, as mentioned in my [comment](https://github.com/stonith404/pingvin-share/issues/318#issuecomment-1801229773) above.

It also would be possible to bypass the the built-in proxy by exposing the port of the backend and frontend. The reverse proxy would still run in the background though.
